### PR TITLE
STORM-3123: Storm Kafka Monitor cannot connect to Kafka over two-way SSL

### PR DESCRIPTION
**Pull request by VipinRathor**
### What is this PR for?
This PR introduces a new configuration option "--consumer-config" (or "-c") in KafkaOffsetLagUtil to connect to Kafka Broker over two-way SSL. New option expects a configuration file containing security properties for Kafka.

Behavior without this PR:
When Kafka broker is running with two-way SSL (i.e. ssl.client.auth=required), Storm Kafka monitor would fail to connect to Kafka with this error:
`Warning: no suitable certificate found - continuing without client authentication
*** Certificate chain
<Empty>
***`

Behavior with this PR:
KafkaOffsetLagUtil can now query offset lag for Kafka spout with 2-way SSL authentication

### What type of PR is it?
[Bug Fix]

### What is the Jira issue?
* https://issues.apache.org/jira/browse/STORM-3123

### How should this be tested?
1. Configure Kafka Broker with 2-way SSL authentication (make sure that ssl.client.auth=required is set in the broker configuration)
2. Create a topic 'testoffset' and produce some messages to it.
3. Run KafkaOffsetLagUtil from command line:
`/usr/jdk64/jdk1.8.0_77/bin/java -cp /root/storm-kafka-monitor-1.1.0.2.6.3.0-SNAPSHOT.jar org.apache.storm.kafka.monitor.KafkaOffsetLagUtil -t testoffset -g testgroup -b node2.openstacklocal:6669 --security-protocol SSL --consumer-config ./consumer.properties`
where the file consumer.properties contains:
ssl.truststore.location=/tmp/kafkaCert/client.truststore.jks
ssl.truststore.password=password
ssl.keystore.location=/tmp/kafkaCert/client.keystore.jks
ssl.keystore.password=password
security.protocol=SSL
4. KafkaOffsetLagUtil would show correct offset log from Kafka topic 'testoffset'.

### CODE REVIEW

**Comment by VipinRathor** [comment]
1/10 Travis CI build failed. Resubmitting to trigger the CI again.

File: external/storm-kafka-monitor/src/main/java/org/apache/storm/kafka/monitor/KafkaOffsetLagUtil.java [hunk 1]
```diff
@@ -145,7 +148,8 @@ public static void main(String args[]) {
                 NewKafkaSpoutOffsetQuery newKafkaSpoutOffsetQuery =
                     new NewKafkaSpoutOffsetQuery(commandLine.getOptionValue(OPTION_TOPIC_LONG),
                                                  commandLine.getOptionValue(OPTION_BOOTSTRAP_BROKERS_LONG),
-                                                 commandLine.getOptionValue(OPTION_GROUP_ID_LONG), securityProtocol);
+                                                 commandLine.getOptionValue(OPTION_GROUP_ID_LONG), securityProtocol,
```
Comment (priyank5485) [inline external/storm-kafka-monitor/src/main/java/org/apache/storm/kafka/monitor/KafkaOffsetLagUtil.java]: I suggest adding a check in the old spout if block and throw an error if this option provided with old spout.

**Comment by priyank5485** [comment]
@VipinRathor You should also update 
https://github.com/apache/storm/blob/master/storm-core/src/jvm/org/apache/storm/utils/TopologySpoutLag.java
and
https://github.com/apache/storm/blob/master/external/storm-kafka-client/src/main/java/org/apache/storm/kafka/spout/KafkaSpout.java

That will make sure that lags will show up in storm ui as well where spout is talking to kafka on SSL listener. Can you also test it?

**Comment by VipinRathor** [comment]
Oops, somehow I ended up pulling the diff commit here in this PR while merging with master. Now it's showing 76 commits. :|
Can this be fixed? If not, I can close this and open a fresh PR.

**Comment by HeartSaVioR** [comment]
@VipinRathor 
You can checkout a new branch based on recent master, and cherry-pick your commits, and reset your PR branch with the last commit of new branch via `git reset --hard`.

**Comment by HeartSaVioR** [comment]
@VipinRathor Any updates? It is completely OK for you to just close an create a new PR.

**Comment by VipinRathor** [comment]
yes, I cherry-picked my 3 commits into another branch and currently testing with Kafka & two-way SSL.

**Comment by HeartSaVioR** [comment]
Could be closed as #2906 supersedes this.

**Pull request closed without merging**